docs/DeveloperGuide.md

```diff
@@ -156,6 +159,45 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### <a name="switching"></a>Switching between tabs
+
+The switching of tabs is facilitated by `SwitchCommand`, `LogicManager`, `MainWindow` and `Tab`. `Tab` is an enum class that represents the various tabs that exist in ATAS.
+
+Given below is an example usage scenario and how the switch mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The default "Students" tab is displayed.
+
+Step 2. The user executes `switch sessions` command to view the sessions tab. `MainWindow#executeCommand()` is called and `LogicManager` calls `AtasParser#parseCommand()`. This results in the creation of a `SwitchCommandParser` to parse the user input.
+
+Step 3. `SwitchCommandParser#parse()` checks if there is an argument being passed by the user. If an argument is passed, a `SwitchCommand` will be created and `SwitchCommand#execute()` will be called by the `LogicManager`.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If no argument is found, a `ParseException` will be thrown and the execution terminates.
+
+</div><br>
+
+Step 4. `SwitchCommand#execute()` will check if the argument passed in is an existing tab name. If the argument is valid, a `CommandResult` containing a variable `switchTab` with the value of the target `Tab` will be return to `MainWindow`.
```

Comment:
Perhaps can combine "If the argument is valid" with no argument ==> If argument is invalid, a `ParseException`...?

Reply:
Thanks for the suggestion but if the argument is invalid, it throws a CommandException but if argument is empty it throws a ParseException. Hence, this is why I separated them.

Reply:
> Thanks for the suggestion but if the argument is invalid, it throws a CommandException but if argument is empty it throws a ParseException. Hence, this is why I separated them.

Noted thanks for clarification